python/ray/serve/_private/application_state.py

```diff
@@ -846,6 +846,8 @@ def shutdown(self) -> None:
         for app_state in self._application_states.values():
             app_state.delete()
 
+        self._kv_store.delete(CHECKPOINT_KEY)
+
     def is_ready_for_shutdown(self) -> bool:
```

Comment:
Non-blocker: Since this is fast enough operation, probably can go by without it. But would be nice to also add a check in `is_ready_for_shutdown()` to ensure the checkpoint is deleted similar to this https://github.com/ray-project/ray/blob/master/python/ray/serve/_private/deployment_state.py#L2495

Reply:
I believe once `delete` returns, the key is deleted

Reply:
Yea, not need to change, just thought to bring it up 🙃